src/app/barter/components/barter-offer/barter-offer.component.ts

```diff
@@ -0,0 +1,83 @@
+import {Component, OnInit, ViewChild} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {User} from '../../../../entities/user.model';
+import {Item} from '../../../../entities/item.model';
+import {getMyItems, ItemsState} from '../../../items/reducers/items.reducer';
+import {takeUntil} from 'rxjs/operators';
+import {ObservableListener} from '../../../components/observable-listener';
+import {Store} from '@ngrx/store';
+import {AuthenticateService} from '../../../user/services/authentication.service';
+import {ItemComponent} from '../../../items/components/item/item.component';
+import {getMyTransactions, TransactionsState} from '../../reducers/transactions.reducer';
+import {createTransaction, createTransactionFail, initiateTransactions} from '../../actions/transactions.actions';
+import {Transaction} from '../../../../entities/transaction.model';
+import firebase from 'firebase';
+
+@Component({
+  selector: 'app-barter-offer',
+  templateUrl: './barter-offer.component.html',
+  styleUrls: ['./barter-offer.component.scss']
+})
+export class BarterOfferComponent extends ObservableListener implements OnInit {
+  user: User;
+  offerItem: Item;
+  myItems: Item[] = [];
+  selectedItem: Item;
+  private myTransactions: Transaction[];
+  @ViewChild('myItem') myItem!: ItemComponent;
+
+
+  constructor(private activatedRoute: ActivatedRoute,
+              private store$: Store<ItemsState>,
+              private transactionsStore$: Store<TransactionsState>,
+              private authService: AuthenticateService) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this.transactionsStore$.dispatch(initiateTransactions());
+    this.transactionsStore$.select(getMyTransactions(this.authService.getUserFromLocalStorage().id))
+      .subscribe(myTransactions => {
+        this.myTransactions = myTransactions;
+      });
+
+
+    this.activatedRoute.data.subscribe((result) => {
+      this.user = result.data[0];
+      this.offerItem = result.data[1];
+      this.store$.select(getMyItems(this.authService.getUserFromLocalStorage().id)).pipe(takeUntil(this.unsubscribeOnDestroy))
+        .subscribe(items => {
+          this.myItems = items;
+          this.selectedItem = items[0];
+        });
+    });
+  }
+
+  selectItem(item: Item) {
+    this.selectedItem = item;
+    this.myItem.changeImageSrc(this.selectedItem.pictureUrls[0]);
+  }
+
+  offerDeal(): void {
+    const invalidTransactions: Transaction[] = this.myTransactions.filter(currentTransaction => !currentTransaction.isTransactionCompleted
```

Comment:
maybe we should save the available transactions in the store instead of filtering it every time?

Reply:
mmm you might right but it's the only place I'm using it- if it will require reuse I will take it out(If I take it to the store it will still filter every time because I need to call it when offerDeal happens)